modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1273,71 +1275,58 @@ private void dropTableLocally(long causalityToken, CatalogTableDescriptor tableD
+        localPartsByTableIdVv.update(causalityToken, (previousVal, e) -> inBusyLock(busyLock, () -> {
+            if (e != null) {
+                return failedFuture(e);
             }
 
-            localPartsByTableIdVv.update(causalityToken, (previousVal, e) -> inBusyLock(busyLock, () -> {
-                if (e != null) {
-                    return failedFuture(e);
-                }
+            var newMap = new HashMap<>(previousVal);
+            newMap.remove(tableId);
 
-                var newMap = new HashMap<>(previousVal);
-                newMap.remove(tableId);
+            return completedFuture(newMap);
+        }));
 
-                return completedFuture(newMap);
-            }));
+        tablesByIdVv.update(causalityToken, (previousVal, e) -> inBusyLock(busyLock, () -> {
+            if (e != null) {
+                return failedFuture(e);
+            }
 
-            tablesByIdVv.update(causalityToken, (previousVal, e) -> inBusyLock(busyLock, () -> {
-                if (e != null) {
-                    return failedFuture(e);
-                }
+            var map = new HashMap<>(previousVal);
 
-                var map = new HashMap<>(previousVal);
+            TableImpl table = map.remove(tableId);
 
-                TableImpl table = map.remove(tableId);
+            assert table != null : tableId;
 
-                assert table != null : tableId;
+            InternalTable internalTable = table.internalTable();
 
-                InternalTable internalTable = table.internalTable();
+            CompletableFuture<?>[] stopReplicaFutures = new CompletableFuture<?>[partitions];
 
-                for (int partitionId = 0; partitionId < partitions; partitionId++) {
-                    closePartitionTrackers(internalTable, partitionId);
-                }
+            for (int partitionId = 0; partitionId < partitions; partitionId++) {
+                var replicationGroupId = new TablePartitionId(tableId, partitionId);
 
-                // TODO: IGNITE-18703 Destroy raft log and meta
+                stopReplicaFutures[partitionId] = stopPartition(replicationGroupId, table);
+            }
 
-                CompletableFuture<Void> destroyTableStoragesFuture = allOf(removeStorageFromGcFutures)
-                        .thenCompose(unused -> allOf(
-                                internalTable.storage().destroy(),
-                                runAsync(() -> internalTable.txStateStorage().destroy(), ioExecutor))
-                        );
+            // TODO: IGNITE-18703 Destroy raft log and meta
+            CompletableFuture<Void> destroyTableStoragesFuture = allOf(stopReplicaFutures)
+                    .thenCompose(unused -> allOf(
+                            internalTable.storage().destroy(),
+                            runAsync(() -> internalTable.txStateStorage().destroy(), ioExecutor))
+                    );
```

Comment:
Seems you could just use `stopAndDestroyPartition` here

Reply:
Using `destroy` is more effective, since you don't have to destroy storages one by one